src/user/screens/starred-repository-list.screen.js

```diff
@@ -1,48 +1,85 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import { FlatList } from 'react-native';
-
-import { getStarredRepositories } from 'user';
+import { FlatList, View, ActivityIndicator } from 'react-native';
 
+import { RestClient } from 'api';
 import {
   ViewContainer,
   RepositoryListItem,
   LoadingRepositoryListItem,
 } from 'components';
 
-const mapStateToProps = state => ({
-  user: state.user.user,
-  starredRepositories: state.user.starredRepositories,
-  isPendingStarredRepositories: state.user.isPendingStarredRepositories,
-});
+const mapStateToProps = (state, ownProps) => {
+  const {
+    auth: { user },
+    pagination: { ACTIVITY_GET_STARRED_REPOS_FOR_USER },
+    entities: { repos },
+  } = state;
+
+  const userId = ownProps.navigation.state.params.user.login;
+
+  const starredReposPagination = ACTIVITY_GET_STARRED_REPOS_FOR_USER[
+    userId
+  ] || {
+    ids: [],
+    isFetching: true,
+  };
+  const starredRepos = starredReposPagination.ids.map(id => repos[id]);
+
+  return {
+    user,
+    starredReposPagination,
+    starredRepos,
+    userId,
+  };
+};
```

Comment:
mapStateToProps now create two props: 

* starredReposPagination: STORE.pagination.ACTIVITY_GET_STARRED_REPOS_FOR_USER['machour'] **or** and empty pagination
* starredRepos: An array of repositories picked from STORE.entities.repos, filtered to contain only those whose id is present in my pagination.

Reply:
Is `starredReposPagination` just an object that returns `nextPageUrl` and an `isFetching` bool?

Reply:
+ `pageCount` and `ids`.

It's literally this in our case:

<img width="572" alt="screen shot 2018-03-26 at 1 03 57 am" src="https://user-images.githubusercontent.com/304450/37881538-a78cd8ba-3091-11e8-9e6f-267b38b776ee.png">